Source/DataSources/MinimumTerrainHeightProperty.js

```diff
@@ -0,0 +1,160 @@
+     *
+     * @example
+     * var hierarchy = new Cesium.ConstantProperty(polygonPositions);
+     * var redPolygon = viewer.entities.add({
+     *     polygon : {
+     *         hierarchy : hierarchy,
+     *         material : Cesium.Color.RED,
+     *         height : 1800.0,
+     *         extrudedHeight : new Cesium.MinimumTerrainHeightProperty(hierarchy)
+     *     }
+     * });
+     */
+    function MinimumTerrainHeightProperty(positions) {
+        this._positions = undefined;
+        this._subscription = undefined;
+        this._definitionChanged = new Event();
+
+        this.positions = positions;
+    }
+
+    defineProperties(MinimumTerrainHeightProperty.prototype, {
+        /**
+         * Gets a value indicating if this property is constant.
+         * @memberof MinimumTerrainHeightProperty.prototype
+         *
+         * @type {Boolean}
+         * @readonly
+         */
+        isConstant : {
+            get : function() {
+                return Property.isConstant(this._positions);
+            }
+        },
+        /**
+         * Gets the event that is raised whenever the definition of this property changes.
+         * @memberof MinimumTerrainHeightProperty.prototype
+         *
+         * @type {Event}
+         * @readonly
+         */
+        definitionChanged : {
+            get : function() {
+                return this._definitionChanged;
+            }
+        },
+        /**
+         * Gets or sets the positions property used to compute the value.
+         * @memberof MinimumTerrainHeightProperty.prototype
+         *
+         * @type {Property}
+         */
+        positions : {
+            get : function() {
+                return this._positions;
+            },
+            set : function(value) {
+                var oldValue = this._positions;
+                if (oldValue !== value) {
+                    if (defined(oldValue)) {
+                        this._subscription();
+                    }
+
+                    this._positions = value;
+
+                    if (defined(value)) {
+                        this._subscription = value._definitionChanged.addEventListener(function() {
+                            this._definitionChanged.raiseEvent(this);
+                        }, this);
+                    }
+
+                    this._definitionChanged.raiseEvent(this);
+                }
+            }
+        }
+    });
+
+    /**
+     * Gets the minimum terrain height based on the positions positions at the provided time.
+     *
+     * @param {JulianDate} time The time for which to retrieve the value.
+     * @returns {Number} The minimum terrain height
+     */
+    MinimumTerrainHeightProperty.prototype.getValue = function(time) {
+        return this._getValue(time);
+    };
+
+    /**
+     * @private
+     */
+    MinimumTerrainHeightProperty.prototype._getValue = function(time) {
```

Comment:
Is there a reason this is not just inline in `getValue`?

Reply:
Yes, because I blindly copied `VelocityVectorProperty` =P
It's probably not needed